packages/manager/apps/key-management-service/src/api/hooks/useKMSServiceInfos.ts

```diff
@@ -0,0 +1,44 @@
+import { useQuery } from '@tanstack/react-query';
+import { getServiceInfos } from '../services';
+import { OKMS } from '@/interface';
+import { ErrorResponse } from '../GET/apiv2/services';
+
+type ServiceLifecycleCurrent = {
+  creationDate: string;
+};
+
+type ServiceLifecycle = {
+  current: ServiceLifecycleCurrent;
+};
+
+type ServiceBilling = {
+  engagement?: string | null;
+  expirationDate?: string | null;
```

Comment:
As it's optional, no need to add the `null` type
```suggestion
  engagement?: string;
  expirationDate?: string;
```

Reply:
When optional it would be defined as undefined.
But the backend can send us a string or null but, maybe, not sending it at all.

I took exemple from somewhere else into the project. Maybe there is legacy conditions leading us to get optional properties.